### DISCUSSION TITLE
v4: Query distinct properties of embedded properties

### DISCUSSION THREAD

**Discussion by Bohne13** [isar/isar#1547] [Help] [answered]
Hello everyone,
I use Isar v4 and do like it! 
In general I'm not understanding how queries work in detail, and didn't found much examples etc. to figure it out by myself.
So maybe someone could help me out!

I have simplified following Isar database classes:

```dart
@collection
class PilotLog {
    final Flight flight;
}

@embedded
class Flight {
    late String flightFrom;
    late String flightTo;
}
```

Is it possible to write a query to get a ```List<String>``` as return with all distinct values of ```flightFrom``` from all ```PilotLog``` records of the database?

I'm really thankful for help and hope to see Isar v4 in a stable version soon.

**Comment by Bohne13** [comment]
Partially I figured it out by myself. How did I figured it out?
Basically, I can say: "Read the f**cking documentation!" 

my sollution for now:

```dart
isar.pilotLogs.where().flightProperty().findAll();
``` 

what I haven't found in the documentation, how to geht down the path to the embedded property of ```flightFrom``` to get a distinct list of it?!?

**Reply by ndelanou** [reply, marked as answer]
Once you have your logs you can simply `final flightFromList = logs.map((log) => log.flightFrom).toList();`

**Reply by Bohne13** [reply]
maybe at the end instead of ```.toList()``` I could use ```.toSet()```, then I have a distinct list/set.

Thanks, @ndelanou I will try this.

I thought, maybe it's possible to do most things already in a single query to only use data I realy need at that moment.

**Reply by ndelanou** [reply]
I don't think it is possible to extract a List of a single field from a query.
This could be a great feature. I needed this several times.

**Reply by Bohne13** [reply]
Indeed, but with my limited knowledge of Isar I think it could be difficult to generate functions for every layer someone might programm, like ```World.Europe.Germany.Bavaria.Munich.getAllStreets()```
